[BugFix] fill AUTO_INCREMENT fail in Cloud Native mode(#26584)

srlch opened the pull request:
Problem:
There are two reasons that will cause a failure when filling AUTO_INCREMENT id in partial update mode for a table in Cloud Native mode:
1. We can not get the tablet meta from the cache and get nullptr crash
2. Currently, seek pk index will fail if the index has not been cached. Filling AUTO_INCREMENT id will fail if there the index has not been cached. But it will keeping failing if no other load is committed and cache the index into memory.

Fixes #26584

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

luohaha commented on be/src/storage/lake/delta_writer.cpp:
```diff
@@ -248,7 +248,7 @@ inline Status DeltaWriterImpl::flush_async() {
         RETURN_IF_ERROR(_mem_table->finalize());
         if (_miss_auto_increment_column && _mem_table->get_result_chunk() != nullptr) {
-            _fill_auto_increment_id(*_mem_table->get_result_chunk());
+            RETURN_IF_ERROR(_fill_auto_increment_id(*_mem_table->get_result_chunk()));
```
If `_fill_auto_increment_id` fail, will it affect `flush_async` ?.

srlch replied:
I think it will affect it. And I think it is the correct behavior if _fill_auto_increment_id fail.

srlch committed:
[BugFix] fill AUTO_INCREMENT fail in Cloud Native mode(#26584)

Problem:
There are two reasons that will cause be fail when filling AUTO_INCREMENT
id in partial update mode for table in Cloud Native mode:
1. We can not get the tablet meta from cache and get nullptr crash
2. Currently, seek pk index will fail if the index has not been cached.
    Filling AUTO_INCREMENT id will fail if there the index has not been cached.
    But it will keep failing if there is no other load commit and cache the
    index into memory.

Signed-off-by: srlch <linzichao@starrocks.com>

wanpengfei-git commented:
@Mergifyio backport branch-3.1

wanpengfei-git commented:
@Mergifyio backport branch-3.0
